src/components/ContentEditor/WidgetLiveEdit/WidgetLiveEdit.tsx

```diff
@@ -0,0 +1,99 @@
+import React, { useState, useEffect, useContext } from 'react';
```

Comment:
I think you are binding too much logic to `WidgetLiveEdit` and the rest of the view logic. You should instead send the editing state to firebase when the edit button is _clicked_-- this would involve you creating a class or function to be called instead of rendering a react component. 

Then you render the react component based on the state instead of updating firebase. 

Essentially delegate all the firebase-related tasks to functions and make the stuff in the view only do view-related logic.